x-pack/platform/packages/shared/onechat/onechat-genai-utils/tools/steps/perform_match_search.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+
+export interface MatchResult {
+  id: string;
+  index: string;
+  highlights: string[];
+}
+
+export interface PerformMatchSearchResponse {
+  results: MatchResult[];
+}
+
+export const performMatchSearch = async ({
+  term,
+  fields,
+  index,
+  size,
+  esClient,
+}: {
+  term: string;
+  fields: string[];
+  index: string;
+  size: number;
+  esClient: ElasticsearchClient;
+}): Promise<PerformMatchSearchResponse> => {
+  const response = await esClient.search<any>({
+    index,
+    size,
+    retriever: {
+      rrf: {
```

Comment:
suggestion: i would group all text fields together into one retriever and then each semantic_text field as individual retrievers. RRF would not be applied if its just all text fields.

Reply:
++. I wanted to keep it as a follow up because it requires `performMatchSearch` to know about field types, and it doesn't at the moment